packages/editor/src/components/post-slug/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { withDispatch, withSelect } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, compose } from '@wordpress/compose';
+import { safeDecodeURIComponent } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import PostSlugCheck from './check';
+import { cleanForSlug } from '../../utils/url';
+
+export class PostSlug extends Component {
+	constructor( { postSlug, postTitle, postID } ) {
+		super( ...arguments );
+
+		this.state = {
+			editedSlug: safeDecodeURIComponent( postSlug ) || cleanForSlug( postTitle ) || postID,
+		};
+
+		this.setSlug = this.setSlug.bind( this );
+	}
+
+	setSlug( event ) {
+		const { postSlug, onUpdateSlug } = this.props;
+		const { value } = event.target;
+
+		const editedSlug = cleanForSlug( value );
+
+		if ( editedSlug === postSlug ) {
+			return;
+		}
+
+		onUpdateSlug( editedSlug );
+	}
+
+	render() {
+		const { instanceId } = this.props;
+		const { editedSlug } = this.state;
+
+		const inputId = 'post-slug-input-' + instanceId;
```

Comment:
We should stick to the `package-directory__descriptor-foo-bar` convention.

Reply:
That's an ID though, not a class.
As far as I can see with a quick search, we don't have any conventions for it. 🙂

Reply:
We use ${className}-${id}.